### PR TITLE
Convert output of a hive run to ps1/sh/http scripts; extract spec/cfg

### PR DESCRIPTION
**Pull request by flcl42**
A tool to transform hive execution results into scripts:

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [x] Other: tool

#### Requires testing

- [ ] Yes
- [x] No

### CODE REVIEW

File: tools/HiveToScripts/Program.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        }
+    }
+    Directory.CreateDirectory(outputDirectory);
+    if (doLog) Console.WriteLine($"Running...");
+    HashSet<string> createdFiles = [];
+
+    using StreamWriter testListWriter = new(Path.Combine(outputDirectory, "__tests.txt"));
+    HashSet<string> reportedContainers = [];
+
+    Regex starter = new("^>>\\s+\\(([a-f0-9]+)\\)\\s+");
```
Comment (ak88) [inline tools/HiveToScripts/Program.cs:51]: Not all hive tests will write the container id like this.

File: tools/HiveToScripts/Program.cs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+if (Directory.Exists(Path.Combine(hiveLogsDirectory, "logs")))
+{
+    hiveLogsDirectory = Path.Combine(hiveLogsDirectory, "logs");
+}
+
+string outputDirectory = args.Length > 1 ? args[1] : "output";
+string jsonrpcUrl = args.Length > 2 ? args[2] : "http://localhost:8551";
+
+string[] simulatorFilePaths = Directory.GetFiles(hiveLogsDirectory, "*-simulator-*.log").Select(f => Path.Combine(outputDirectory, f)).ToArray();
```
Comment (ak88) [inline tools/HiveToScripts/Program.cs:30]: Could it be an idea to parse the json file first for failed tests, and then use the client info id to search for the relevant requests? That way you could categorise tests under folders with a shortened test name perhaps.

File: tools/HiveToScripts/Program.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+foreach (string simulatorFilePath in simulatorFilePaths)
+{
+    using StreamReader simulatorFile = File.OpenText(simulatorFilePath);
+
+    if (Directory.Exists(outputDirectory))
+    {
+        foreach (string f in Directory.EnumerateFiles(outputDirectory))
+        {
+            File.Delete(f);
```
Comment (ak88) [inline tools/HiveToScripts/Program.cs:41]: This will attempt to delete files which you still have handles to in `fileDescriptors `

Comment (flcl42) [reply tools/HiveToScripts/Program.cs:41]: Yeah, should be done outside of the foreach

File: tools/HiveToScripts/Program.cs (hunk 1 above)
Comment (flcl42) [reply tools/HiveToScripts/Program.cs:51]: Which ones, could you share example? I've never faced such problem

File: tools/HiveToScripts/Program.cs (hunk 2 above)
Comment (flcl42) [reply tools/HiveToScripts/Program.cs:30]: Good idea! It'll require read it two times, but also allows to ignore successful tests!

File: tools/HiveToScripts/Program.cs (hunk 1 above)
Comment (ak88) [reply tools/HiveToScripts/Program.cs:51]: > Which ones, could you share example? I've never faced such problem

I know rpccompat  for one, i am not sure if there are more

Comment (flcl42) [reply tools/HiveToScripts/Program.cs]: Fixed

**Pull request closed without merging**